docs/sections/workflows/workchain.md

```diff
@@ -613,101 +621,149 @@ Out[3]:
-The resulting code looks like this:
+Of course, the power of a work chain lies in its ability to _submit_ other processes that can run independently while the work chain waits for them to complete.
+Doing so also releases the daemon to do other tasks, which is vital when running many workflows in high-throughput.
+These processes often don't run Python code - think for example of a remote binary running on a supercomputer.
 
-```{literalinclude} include/code/workchain/my_first_workchain_6_calcjob.py
-:language: python
-:emphasize-lines: 1-5, 24, 41-46, 48, 54
-```
+Although a work chain can also submit other work chains, in this section we'll see how to submit a _calculation job_ ({class}`~aiida.engine.processes.calcjobs.calcjob.CalcJob`) inside a work chain. Starting from the `AddWorkChain` in the {ref}`section on work chain context <workflows-workchain-context>` the code below replaces the `addition` calculation function by the `ArithmeticAdd` calculation job, which ships with `aiida-core`:
 
-Let us break down what we did.
-First, we imported the {class}`~aiida.engine.processes.calcjobs.calcjob.CalcJob` that we plan on using with the help of the `CalculationFactory`:
+:::{margin}
+{{ download }} **{download}`Download the script! <include/code/workchain/addcalcjobworkchain.py>`**
+:::
 
-```{literalinclude} include/code/workchain/my_first_workchain_6_calcjob.py
+```{literalinclude} include/code/workchain/addcalcjobworkchain.py
 :language: python
-:lines: 5
+:emphasize-lines: 1-5, 17, 24-30, 32, 38
 ```
 
-We also declared a new input, which is a code:
+Let's have a closer look at each change in the code.
+First, we imported the `ArithmeticAddCalculation` calculation job that we plan on using with the help of the `CalculationFactory`:
 
-```{literalinclude} include/code/workchain/my_first_workchain_6_calcjob.py
+```{literalinclude} include/code/workchain/addcalcjobworkchain.py
 :language: python
-:lines: 24
+:lines: 3-5
 ```
 
-(**TODO: Somebody needs to try and explain this idea of passing a code as an input, then running a calc job process which executes the code...**)
+However, it is possible to set up multiple codes in the AiiDA database that run the _same_ calculation job (a local and remote one, for example).
+Hence, the user must also be able to specify _which code_ the work chain should run.
+To allow this, we have added a `code` input to the work chain `spec`, which must be of type `Code`:
 
+```{literalinclude} include/code/workchain/addcalcjobworkchain.py
+:language: python
+:lines: 17
+```
 
-In the `add()` method, which is the second step in the outline of the work chain, instead of a calculation function, we are submitting the {class}`~aiida.engine.processes.calcjobs.calcjob.CalcJob` named `ArithmeticAddCalculation`:
+In the `add()` method we now _submit_ the `ArithmeticAddCalculation` calculation _job_ instead of _running_ the `addition()` calculation _function_:
```

Comment:
```suggestion
In the `add()` method, we now _submit_ the `ArithmeticAddCalculation` calculation _job_ instead of _running_ the `addition()` calculation _function_:
```